ZTNA/ZTNA-FortiGate.yaml

```diff
@@ -0,0 +1,682 @@
+          - - |
+              {
+            - '"bucket"'
+            - ' : "'
+            - !Ref InitS3Bucket
+            - '"'
+            - |
+              ,
+            - '"region"'
+            - ' : '
+            - '"'
+            - !Ref 'AWS::Region'
+            - '"'
+            - !If 
+                - LicenseTypeBYOL
+                - 
+                  !Join
+                    - ''
+                    - - |
+                        ,
+                      - '"license" : "'
+                      - !Ref ZTNAGatewayLicenseFile
+                      - '"'
+                - ''
+            - |
+              ,
+            - '"config"'
+            - ' : '
+            - '"'
+            - /fgt.txt"
+            - |+
+
+            - '}'
+  fgt1eni0:
+    Type: 'AWS::EC2::NetworkInterface'
+    Properties:
+      Description: port1
+      GroupSet:
+        - !Ref ZTNAGatewaySecGrp
+      SourceDestCheck: false
+      SubnetId: !Ref PublicSubnet
+      Tags:
+        - Key: Name
+          Value: !Join ['', [!Ref 'AWS::StackName', -fgt1eni0]]
+        - Key: Interface
+          Value: eth0
+      PrivateIpAddresses:
+        - PrivateIpAddress: !Select ['0', !Split [/, !Ref ZTNAGatewayPublicIP]]
+          Primary: true
+  FGTEIP:
+    Type: 'AWS::EC2::EIP'
+    Properties:
+      Domain: vpc
+  FGTIPASSOCIATION:
+    Type: 'AWS::EC2::EIPAssociation'
+    Properties:
+      AllocationId: !GetAtt [FGTEIP, AllocationId]
+      NetworkInterfaceId: !Ref fgt1eni0
+      PrivateIpAddress: !Select ['0', !Split [/, !Ref ZTNAGatewayPublicIP]]
+    DependsOn: FGTEIP
+  fgt1eni1:
+    Type: AWS::EC2::NetworkInterface
+    Properties:
+      Description: port2
+      GroupSet:
+      - !Ref 'ZTNAGatewaySecGrp'
+      SourceDestCheck: false
+      SubnetId: !Ref 'PrivateSubnet'
+      Tags:
+      - Key: Name
+        Value: !Join ['', [!Ref 'AWS::StackName', -fgt1eni1]]
+      PrivateIpAddress: !Select ['0', !Split [/, !Ref 'ZTNAGatewayPrivateIP']]
+  LambdaRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Path: /
+      Policies:
+        - PolicyName: S3AccessRole
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 's3:PutObject'
+                Resource: !Sub ['arn:aws:s3:::${Bucketname}/fgt.txt', {Bucketname: !Ref 'InitS3Bucket'}]
+              - Effect: Allow
+                Action:
+                  - 'logs:CreateLogGroup'
+                  - 'logs:CreateLogStream'
+                  - 'logs:PutLogEvents'
+                Resource: 'arn:aws:logs:*:*:*'
+  InitFunction:
+    Type: 'AWS::Lambda::Function'
+    Properties:
+      Code:
+        ZipFile: !Join 
```

Comment:
In the future you may want to do the lambda config as a file in a bucket instead so it can be adjusted without modifying the lambda code.